boards/arm64/qemu_cortex_a53/qemu_cortex_a53.yaml

```diff
@@ -0,0 +1,46 @@
+identifier: qemu_cortex_a53
+name: QEMU Emulation for Cortex-A53
+type: qemu
+simulation: qemu
+arch: arm64
+toolchain:
+  - cross-compile
+ram: 128
+testing:
+  default: false
+  ignore_tags:
+    - base64
```

Comment:
That list is actually taken from `boards/arm/qemu_cortex_r5/qemu_cortex_r5.yaml`.

This is not a full port, I have something more complete in my working repo but my idea was trying to upstream as gradually as possible. Is it possible to define the minimum set of tests to pass to have something merged? It's a new, hard architecture and upstreaming all at once can be a bit overwhelming.

Reply:
> That list is actually taken from boards/arm/qemu_cortex_r5/qemu_cortex_r5.yaml.

Ugh, that should not have gone in.

> Is it possible to define the minimum set of tests to pass to have something merged?

I think you will find that most things will start passing once you get a working timer. You don't have a timer right now and that is a major gap. All you generally need to pass tests is a working timer and serial port.

>  It's a new, hard architecture and upstreaming all at once can be a bit overwhelming.

In what way? Did you have to touch a lot of code outside of arch/?

Reply:
> upstream as gradually as possible

We like code to go in when it's backed up by tests that prove that it works.
Consider a case where someone contributes a lot of code, that isn't backed up by tests, and promises are given that passing tests will come later. That person then proceeds to disappear off the face of the Earth. It's happened a few too many times to me that I must insist on seeing some proof that what we have here works before it can go in. I don't think you're that far off once you get the timer going.

Reply:
> I think you will find that most things will start passing once you get a working timer. You don't have a timer right now and that is a major gap.

Just a quick comment regarding this. `qemu_cortex_a53` here is currently using virt machine that has no real hardware equivalent.

I think it would benefit from targeting xilinx_zynqmp instead, since that SoC embeds a Cortex-A53 and there is already a functional timer driver for it (i.e. Cadence TTC; it has been verified to be working on the RPU using Cortex-R) and also can be tested and verified against a real hardware platform.

All the other qemu_cortex_* emulated boards are targeting a qemu machine that has a real hardware equivalent too.

Reply:
> I think you will find that most things will start passing once you get a working timer. You don't have a timer right now and that is a major gap. All you generally need to pass tests is a working timer and serial port.

I have a working timer downstream (the ARM generic timer) that I can try to push into the initial release. But having a working timer means hooking up the `GIC`, `isr_wrapper`, basic threading and other things that I wanted to push at a later time and step-by-step because I wanted to focus now on how to integrate the new architecture in the current code-base. Nonetheless I agree with your point.

> We like code to go in when it's backed up by tests that prove that it works.

FWIW the code works in the sense that the only sample that it can actually run (`hello_world`) runs successfully so I can prove that my code works :P

Reply:
@andrewboie asking this again because I find the test system a bit confusing: what tests do you want to see passing to consider the patch-set for inclusion?

I have timer, multi-threading (no preemption), IRQ offloading and all the glue code working on my branch, but it's hard to say what tests can be actually passed with this set of features besides probably `timer_monotonic` and `timer_api` (that I'm able to pass).

I want to postpone implementing preemption to a later time when the basics are already discussed and possibly merged since it is already a fair number of lines of code to review (and because I need more time for that).

Reply:
> @andrewboie asking this again because I find the test system a bit confusing: what tests do you want to see passing to consider the patch-set for inclusion?

Showing that everything in tests/kernel passes (aside from stuff that requires optional features like userspace or SMP) would be a great start.

> I want to postpone implementing preemption to a later time when the basics are already discussed and possibly merged since it is already a fair number of lines of code to review (and because I need more time for that).

Not thrilled with this plan, I'd rather you just got preemption working. You have a timer now, you shouldn't have to write a ton more code, just fix whatever issues are preventing preemption from working.